pallets/bfc-staking/src/pallet/impls.rs

```diff
@@ -301,53 +301,67 @@ impl<T: Config> Pallet<T> {
 		<DelayedPayouts<T>>::insert(round_to_payout, payout);
 	}
 
-	/// Handle validators auto-compoundable round rewards
-	/// If the reward destination is set to `Staked`, it will be auto-compounded
-	pub fn handle_validator_auto_compounding(
-		validator: T::AccountId,
+	/// Handle validators auto-compoundable round rewards payout. If the reward destination is set
+	/// to `Staked`, it will be auto-compounded
+	pub fn handle_validator_reward_payout(
+		controller: T::AccountId,
+		stash: T::AccountId,
 		reward: BalanceOf<T>,
 	) -> Result<(), DispatchError> {
-		if let Some(mut validator_state) = <CandidateInfo<T>>::get(&validator) {
+		if let Some(mut validator_state) = <CandidateInfo<T>>::get(&controller) {
+			// mint rewards to the validators stash account
+			Self::mint_reward(reward, stash);
 			// increment the awarded tokens of this validator
 			validator_state.increment_awarded_tokens(reward);
 
+			// auto-compound round rewards if `reward_dst` is set to `Staked`
 			if validator_state.reward_dst == RewardDestination::Staked {
 				validator_state.bond_more::<T>(
 					validator_state.stash.clone(),
-					validator.clone(),
+					controller.clone(),
 					reward,
 				)?;
-				Self::update_active(&validator, validator_state.voting_power);
+				Self::update_active(&controller, validator_state.voting_power);
 				Self::sort_candidates_by_voting_power();
 			}
-			<CandidateInfo<T>>::insert(&validator, validator_state);
+			<CandidateInfo<T>>::insert(&controller, validator_state);
 		}
 		Ok(().into())
 	}
 
-	/// Handle nominators auto-compoundable round rewards
-	/// If the reward destination is set to `Staked`, it will be auto-compounded
-	pub fn handle_nominator_auto_compounding(
-		validator: T::AccountId,
+	/// Handle nominators auto-compoundable round rewards payout. If the reward destination is set
+	/// to `Staked`, it will be auto-compounded
+	pub fn handle_nominator_reward_payout(
+		controller: T::AccountId,
 		nominator: T::AccountId,
 		reward: BalanceOf<T>,
 	) -> Result<(), DispatchError> {
 		if let Some(mut nominator_state) = <NominatorState<T>>::get(&nominator) {
-			match nominator_state.reward_dst {
-				RewardDestination::Staked => {
-					// increment the awarded tokens of this nominator
-					nominator_state.increment_awarded_tokens(&validator, reward);
-					// auto-compound nomination
-					if nominator_state.increase_nomination::<T>(validator.clone(), reward).is_ok() {
+			// the nominator must not be revoking and leaving
+			if !nominator_state.is_revoking(&controller) && !nominator_state.is_leaving() {
```

Comment:
```suggestion
			if nominator_state.is_active() && !nominator_state.is_revoking(&controller) {
```
형태가 낫지 않을까요? 나중에 NominatorStatus 새로운 것 추가될 수도 있으니까.